route-rs-runtime/src/io/pcap.rs

```diff
@@ -0,0 +1,212 @@
+            }
+            PcapReplayState::RunRealtime(state) => {
+                let peek_intime = if let Some(p) = self.capture.peek() {
+                    (state.start.elapsed() > (p.timestamp - state.cap_start))
+                } else {
+                    //End of Pcap File
+                    return Ok(Async::Ready(None));
+                };
+                if peek_intime {
+                    if let Some(p) = self.capture.next() {
+                        let output_packet: Vec<u8> = p.data;
+                        println!(
+                            "Sending Time {:?} {:?} {:?}",
+                            state.start.elapsed(),
+                            p.timestamp,
+                            p.timestamp - state.cap_start
+                        );
+                        Ok(Async::Ready(Some(output_packet)))
+                    } else {
+                        unreachable!() // This is unexpected as logically peek believes a packet exists
+                    }
+                } else {
+                    // No packet to send reschedule
+                    // TODO: Should we rely on Delay instead and let that notify?
+                    task::current().notify();
+                    Ok(Async::NotReady)
+                }
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::utils::test::harness::run_link;
+    use std::path::PathBuf;
+
+    #[test]
+    fn read_packets_router() {
+        // pcap captures 11 pings
+        let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        d.push("src/utils/test/fixture/test_icmp_8888.pcapng");
+        let file = d.to_string_lossy().into_owned().to_string();
+        println!("Using {}", file);
+
+        let (run0, egr0) = FromPcapProducer::new()
+            .src_file(file)
+            .replay_mode(PcapReplayMode::Dump)
+            .build_link();
+        let link = (run0, egr0);
+        let results = run_link(link);
+        let pkt_request = vec![
+            224, 203, 188, 35, 94, 173, 0, 36, 155, 33, 136, 16, 8, 0, 69, 0, 0, 84, 224, 38, 0, 0,
+            128, 1, 9, 194, 192, 168, 128, 8, 8, 8, 8, 8, 8, 0, 195, 196, 8, 60, 0, 24, 46, 76,
+            193, 93, 0, 0, 0, 0, 113, 106, 12, 0, 0, 0, 0, 0, 16, 17, 18, 19, 20, 21, 22, 23, 24,
+            25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46,
+            47, 48, 49, 50, 51, 52, 53, 54, 55,
+        ];
+        let pkt_reply = vec![
+            0, 36, 155, 33, 136, 16, 224, 203, 188, 35, 94, 173, 8, 0, 69, 0, 0, 84, 0, 0, 0, 0,
+            53, 1, 52, 233, 8, 8, 8, 8, 192, 168, 128, 8, 0, 0, 203, 196, 8, 60, 0, 24, 46, 76,
+            193, 93, 0, 0, 0, 0, 113, 106, 12, 0, 0, 0, 0, 0, 16, 17, 18, 19, 20, 21, 22, 23, 24,
+            25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46,
+            47, 48, 49, 50, 51, 52, 53, 54, 55,
+        ];
+        for i in 0..22 {
```

Comment:
This feels clunky.  Can you put a comment here explaining what is going on?  Especially the mod 2 for each request or reply.  Also why the many asserts rather than just `assert_eq!(results[0][i], packet_request)`?